examples/AdvancedOperations/AddDynamicSearchAds.cs

```diff
@@ -0,0 +1,320 @@
+            AdGroupAd adGroupAd = new AdGroupAd()
+            {
+                AdGroup = adGroupResourceName,
+                Status = AdGroupAdStatus.Paused,
+
+                // Set the ad as an expanded dynamic search ad.
+                Ad = new Ad()
+                {
+                    ExpandedDynamicSearchAd = new ExpandedDynamicSearchAdInfo()
+                    {
+                        Description = "Buy tickets now!"
+                    }
+                }
+            };
+
+            // Create the operation.
+            AdGroupAdOperation operation = new AdGroupAdOperation()
+            {
+                Create = adGroupAd
+            };
+
+            // Add the dynamic search ad.
+            MutateAdGroupAdsResponse response = adGroupAdService.MutateAdGroupAds(
+                customerId.ToString(), new AdGroupAdOperation[] { operation });
+
+            // Display the response.
+            Console.WriteLine($"Added ad group ad with resource name " +
+                $"'{response.Results[0].ResourceName}'.");
+        }
+
+        /// <summary>Add webpage targeting criteria for the DSA ad group.</summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="adGroupResourceName">The resource name of the ad group for which targeting
+        /// is aded.</param>
+        private static void AddWebPageCriteria(GoogleAdsClient client, long customerId,
+            string adGroupResourceName)
+        {
+            // Get the AdGroupCriterionService.
+            AdGroupCriterionServiceClient adGroupCriterionService =
+                client.GetService(Services.V1.AdGroupCriterionService);
+
+            // Create the criterion.
+            AdGroupCriterion adGroupCriterion = new AdGroupCriterion()
+            {
+                AdGroup = adGroupResourceName,
+                CpcBidMicros = 1_000_000,
+                Status = AdGroupCriterionStatus.Paused,
+
+                // Set the webpage targeting criteria.
+                Webpage = new WebpageInfo()
+                {
+                    CriterionName = "Special Offers",
+                    Conditions =
+                    {
+                        // Adds the url targeting criteria.
+                        new WebpageConditionInfo()
+                        {
+                            Operand = WebpageConditionOperand.Url,
+                            Argument = "/specialoffers"
+                        },
+                        // Adds the page title criteria.
+                        // The list of webpage targeting conditions are
+                        // and-ed together when evaluated for targeting.
+                        new WebpageConditionInfo()
+                        {
+                                                        Operand = WebpageConditionOperand.PageTitle,
```

Comment:
The extra space on this line looks a little odd. Is that intentional?

Reply:
Done.